src/compilation/scope_manager.rs

```diff
@@ -3,18 +3,88 @@ use crate::parsing::ast::FuncProto;
 use llvm_sys::prelude::LLVMValueRef;
 use std::collections::HashMap;
 
+/// Struct to store a variable's [Flick type](Type) as well as the corresponding [LLVM object][a].
+///
+/// [a]: LLVMValueRef
 #[derive(Copy, Clone, Debug)]
 pub struct Var {
+    // TODO(tbreydo): Minor: do we need to derive Copy? why is it inconsistent with Func below?
     pub var_type: Type,
     pub value: LLVMValueRef,
 }
 
+/// Struct to store a function's [Flick prototype](FuncProto) as well as the corresponding
+/// [LLVM object][a].
+///
+/// [a]: LLVMValueRef
 #[derive(Clone, Debug)]
 pub struct Func {
     pub proto: FuncProto,
     pub value: LLVMValueRef,
 }
 
+/// This data structure manages namespaces/scopes for variables and functions.
+///
+/// For example, in the following code, there are two scopes, and it would be nice to
+/// have an easy way to manage them during compilation.
+///
+/// ```text
+/// i64 a = 3      // outer scope
+/// if foo() {     // inner scope
+///     i64 a = 2  // inner scope
+/// }              // inner scope
+/// print(a)       // outer scope
+/// ```
+///
+/// Also, giving a variable the same name as a function is silly but should still work:
```

Comment:
Completely silly.... maybe to the point we should disallow it?

Reply:
Added an inline todo with d0028d9. We should def look at this.